src/movie/task/MovieProcessingDemonstrator.java

```diff
@@ -0,0 +1,71 @@
+package movie.task;
+
+import functionalinterface.example.Randomable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class MovieProcessingDemonstrator {
+
+    public static void demonstrateMovieProcessing() {
+        List<Movie> movies = generateMovies(99);
+
+        Map<String, List<Movie>> gouredMovies = movies.stream()
+                .peek(System.out::println)
+                .collect(Collectors.groupingBy(Movie::getGenre));
+        double horrorAveragePrice = gouredMovies.get("Horror").stream()
+                .mapToInt(Movie::getPrice)
+                .average()
+                .getAsDouble();
+        double comedyAveragePrice = gouredMovies.get("Comedy").stream()
```

Comment:
what about using `Team` enum?

Reply:
I don`t think, that it`s necessary here, it won`t improve readability of code and I don`t see any meaningful reasons to use enum, so if it`s a recommendation,but not a requirement I won`t do use enum here.

Reply:
You know what is enum and for what reasons it was created. For representing a group of constants, which keeps in one place. 

What will be if you change `Action` to `Adventure` (for example)? Yep, you will need to change the string in all the places where you use it. 

What will be if you make a mistype in some file and type `Coomedy` instead of `Comedy`? Exactly, something will break.

So, the next decision is yours 😉

Reply:
refactored using enums